src/Buffer.cpp

```diff
@@ -106,11 +106,8 @@ EXPORT RefCount &ref_count<BufferContents>(const BufferContents *p) {
 
 template<>
 EXPORT void destroy<BufferContents>(const BufferContents *p) {
-    // Free any device-side allocation
-    if (p->source_module.free_dev_buffer) {
-        int error = p->source_module.free_dev_buffer(NULL, const_cast<buffer_t *>(&p->buf));
-        user_assert(!error) << "Failed to free device buffer\n";
-    }
+    int error = p->source_module.dev_free(const_cast<buffer_t *>(&p->buf));
```

Comment:
I know this isn't strictly part of this PR, but, it seems like destroy<> should take a non-const pointer.